### DISCUSSION TITLE
How well does AIS-Catcher scale ?

### DISCUSSION THREAD

**Discussion by PideaLtd** [jvde-github/AIS-catcher#219] [General]
I have a site that's running AIS-Catcher version v0.54-26-gab7eadbf which receives messages via UDP that I aggregate from around a dozen receivers.  It's currently displaying 1,204 vessels and the interface is very slow to load and dragging the map takes ages.

Is there any way to improve the responsiveness at all e.g. turning off some components of it or is AIS-Catcher not suited to displaying large numbers of vessels ?

I love the interface but as I add more receivers it don't want it to slow down further.

Thanks

Phil

**Comment by jvde-github** [comment]
Fair observation. This is purely an issue with how the ships and tooltips are plotted on the maps. in the webviewer (and perhaps the refresh frequency can be reduced somewhat from every 2.5s). The current map is not optimized at all so will work well for typical stations but when the number of vessels grows it gets slow when dragging. It already happens with >250 ships. There are a few optimizations that can be done and I am looking at how to best resolve (including to switching to openlayers from leaflet). 

One thing that helps for now is to hide the ship labels, you can also reduce the data refresh but that requires a plugin. I was planning in the next version to do some quick fixes and then a proper rework in the coming months.  Would be nice if you would be able to test that in your set up.

**Comment by jvde-github** [comment]
I already made some tweaks and added that to the map which make a big difference when the labels are off. It is more or less smooth with ~3000 ships on the map. 

The trick was to create tooltips only when needed on screen and block some activity if the user is interacting with the screen. Hope this will help you for now.

**Comment by PideaLtd** [comment]
That's great Jasper - thanks very much for that.  I'd be more than happy to test the new versions.

If I clone the current version and build it will it include the changes ?

Phil

**Comment by jvde-github** [comment]
Yes, indeed, just a fresh normal build of the usual.  This is the version:
![Screenshot from 2024-01-13 12-00-27](https://github.com/jvde-github/AIS-catcher/assets/52420030/04a79228-4e42-4b5e-856f-baee791c4d7d)

Hope you will already see an improvement without permanent labels. Let me know. Cheers, Jasper

**Comment by jvde-github** [comment]
ps you can also see it in action here on a larger set: https://aiscatcher.org/all

**Comment by PideaLtd** [comment]
Brilliant - I've just downloaded and built the latest version on my fairly modest server and it's working much better with 1,213 vessels currently showing.

Thanks for that !  I'll be happy to test any additional changes as I'm adding more feeds all the time.

Phil

**Reply by jvde-github** [reply]
FYI, am looking into scaling for a network version of the software (just neater if you have a network of receivers).

First results with 50K plus vessels at 500 msg/second: https://aiscatcher.org/#liveais

Combination of drawing on the canvas and server side techniques.

**Reply by PideaLtd** [reply]
Ooh, looks really good :-)  Happy to test anything you need although I'm still only at around 1,000 vessels.

Phil

**Comment by Taxom** [comment]
Hi Jasper, I tried something similar with receiving data from AIShub on previous versions but it works too slow. 
I am glad to hear that it works well now.
Can I filter out the data received from TCP (AIShub) to avoid its re-translation to aggregators?  
Thank you!
MaxPi.

**Reply by jvde-github** [reply]
Yes, note these performance enhancement is not yet available, but indeed. If you run let's say two input sources `-d -t txt data.aishub.net`  you can send the input from only the firstto a UDP channel by using `-u 192.168.1.123 4002 GROUPS_IN 1` for the second use `GROUPS_IN 2`  or from both (default ) `GROUPS_IN 1,2`. Not really documented.....

**Comment by jvde-github** [comment]
> Ooh, looks really good :-) Happy to test anything you need although I'm still only at around 1,000 vessels.
> 
> Phil

There is now a fairly complete version of the program based on openlayers, clone with `git clone .... -b openlayers` and build as usual. 
The dashboards on aiscatcher.org that are running on that server also have these changes. 

It runs smoothly with >1000 ships but the map has slightly changed, mostly with the labels.

Comments or bug reports welcome. I have not included in the main branch yet to test a bit more first.
